### PR TITLE
[cloud-provider-azure] add tags support for azure additional storageclasses

### PR DESCRIPTION
**Pull request by sprait**
## Description

Add tag support for additional StorageClasses and fix available values for `storageClass.provision.type`.

## Why do we need it, and what problem does it solve?

To be able to use tags in additional StorageClasses.

## What is the expected result?

ModuleConfig
image

storageClass
image

pv
image
image

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-azure
type: feature
summary: Add tags support for additional StorageClasses.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/azure'...

**Comment by sprait** [comment]
/e2e/abort azure;Standard;containerd;1.25 8734184813 8734184813-1-con-1-25 /sys/deckhouse-oss/install:pr8175 azureuser@20.82.96.70

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/azure'...

File: modules/030-cloud-provider-azure/openapi/config-values.yaml [hunk 1]
```diff
@@ -64,3 +59,9 @@ properties:
               type: number
               description: |
                 Disk throughput in `MBps` (limit of a single disk is 256 KiB/s for each provisioned IOPS).
+            tags:
+              type: string
```
Comment (name212) [inline modules/030-cloud-provider-azure/openapi/config-values.yaml:63]: In configuration we should use object with additionalFields and convert to storage class format in the hook

**Review by name212 — CHANGES_REQUESTED** [review]
one nit

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/azure'...

File: modules/030-cloud-provider-azure/openapi/config-values.yaml (hunk 1 above)
Comment (sprait) [reply modules/030-cloud-provider-azure/openapi/config-values.yaml]: Done, check again pls

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — APPROVED** [review]

File: modules/030-cloud-provider-azure/openapi/config-values.yaml [hunk 2]
```diff
@@ -64,3 +59,22 @@ properties:
...
+                    type: string
+                    pattern: '^[A-Za-z0-9][-A-Za-z0-9_.]*$'
+              x-examples:
+                -
+                  - key: key1
+                    value: val1
+                  - key: key2
+                    value: val2
+              description: |
+                Azure disk [tags](https://cloud-provider-azure.sigs.k8s.io/topics/tagging-resources/)
```
Comment (z9r5) [inline modules/030-cloud-provider-azure/openapi/config-values.yaml:80]:
```suggestion
                Azure disk [tags](https://cloud-provider-azure.sigs.k8s.io/topics/tagging-resources/).
```

File: modules/030-cloud-provider-azure/openapi/doc-ru-config-values.yaml [hunk 3]
```diff
@@ -35,3 +35,6 @@ properties:
             diskMBpsReadWrite:
               description: |
                 Скорость обращения к диску в `MBps` (лимит 256 KiB/s на каждый IOPS).
+            tags:
+              description: |
+                Azure disk [теги](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources)
```
Comment (z9r5) [inline modules/030-cloud-provider-azure/openapi/doc-ru-config-values.yaml:40]:
```suggestion
                Azure disk [теги](https://learn.microsoft.com/ru-ru/azure/azure-resource-manager/management/tag-resources).
```

**Review by z9r5 — APPROVED** [review]

File: modules/030-cloud-provider-azure/openapi/doc-ru-config-values.yaml (hunk 3 above)
Comment (z9r5) [reply modules/030-cloud-provider-azure/openapi/doc-ru-config-values.yaml:40]: @sprait Links to different resources for the en and ru versions.  Let's fix?

**Pull request merged**